src/internal/utils/validators.ts

```diff
@@ -18,6 +18,23 @@ export function validateSortedSetName(name: string) {
   }
 }
 
+export function validateSortedSetIndices(
+  start_index: number,
+  end_index?: number
+) {
+  if (end_index === undefined) {
+    return;
+  }
+  if (start_index > 0 && end_index > 0 && start_index >= end_index) {
+    throw new InvalidArgumentError('start index must be less than end index');
+  }
+  if (start_index < 0 && end_index < 0 && start_index >= end_index) {
```

Comment:
What if start and end are both 0?

Reply:
fair

Reply:
hrm actually what are you asking for here?  end index is exclusive so 0,0 wouldn't really make a lot of sense would it?

Reply:
I thought you were checking for runtime zero-length spans as well. If that's not an issue then we don't need to check for start == end